jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/FileID.java

```diff
@@ -149,35 +189,10 @@ public static boolean isArchive(String filename)
      */
     public static boolean isArchive(URI uri)
     {
-        if (uri == null)
-            return false;
-        if (uri.getScheme() == null)
+        String ext = getExtension(uri);
+        if (ext == null)
             return false;
-        if (uri.getScheme().equalsIgnoreCase("jar"))
-        {
-            URI sspUri = URI.create(uri.getRawSchemeSpecificPart());
-            if (!sspUri.getScheme().equalsIgnoreCase("file"))
-            {
-                return false; // not a `jar:file:` based URI
-            }
-
-            String path = sspUri.getPath();
-
-            int jarEnd = path.indexOf("!/");
-            if (jarEnd >= 0)
-            {
-                return isArchive(path.substring(0, jarEnd));
-            }
-            return isArchive(path);
-        }
-        String path = uri.getPath();
-        // look for `!/` split
-        int jarEnd = path.indexOf("!/");
-        if (jarEnd >= 0)
-        {
-            return isArchive(path.substring(0, jarEnd));
-        }
-        return isArchive(path);
+        return (ext.equals(".jar") || ext.equals(".war") || ext.equals(".zip"));
```

Comment:
Don't you want to use `.equalsIgnoreCase()`?

Reply:
`getExtension()` returns it in lowercase already.